### PR TITLE
ci: remove old vendoring solution in favor of relying on yarn-path

### PR DESCRIPTION
**Pull request by josephperrott**
Now that bazel respects the yarn-path value found in .yarnrc, we can
remove the last remaining reliances on our vendoring in
//third_party/github.com/yarnpkg/yarn/

### CODE REVIEW

**Review by gregmagolan — APPROVED** [review]
So nice when a PR just removes unnecessary complication 👍 👍

**Review by IgorMinar — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -621,11 +603,6 @@ jobs:
           command: |
             git fetch origin $CI_STABLE_BRANCH
             git checkout --force origin/$CI_STABLE_BRANCH -- aio/ .yarn/ .yarnrc
-        # Overwrite yarn again to use the version from the checked out branch.
-      - overwrite_yarn
-        # Ignore yarn's engines check, because we checked out `aio/package.json` from the stable
-        # branch and there could be a node version skew, which is acceptable in this monitoring job.
-      - run: yarn config set ignore-engines true
```
Comment (gkalpak) [inline .circleci/config.yml]: This line (and the associated comments above) is still required.
(NOTE: This is about ignore the check for the Node.js version.)

**Review by gkalpak — APPROVED** [review]
One comment that needs to be addressed.
Otherwise ❤️ 💯 😍 🍾 🎊

File: .circleci/config.yml (hunk 1 above)
Comment (IgorMinar) [reply .circleci/config.yml]: I pushed a fixup commit for this and rebased the PR

**Pull request closed without merging**